include/boost/beast/websocket/impl/close.ipp

```diff
@@ -125,31 +123,26 @@ operator()(
     BOOST_ASIO_CORO_REENTER(*this)
     {
         // Maybe suspend
-        if(! d.ws.wr_block_)
+        if(d.ws.wr_block_.try_lock(this))
```

Comment:
What if paused_close_.emplace() (line 136) throws an exception?

Reply:
It should propagate to the thread calling `io_context::run` like any other exception

Reply:
That's right, if an exception occurs in a composed operation the only subsequent permissible operation on the stream is destruction.